plugins/automox/icon_automox/util/api_client.py

```diff
@@ -0,0 +1,318 @@
+        """
+        return self._call_api("POST", f"{self.endpoint}/servergroups",
+                              params=self._org_param(org_id), json_data=payload)
+
+    def update_group(self, org_id: int, group_id: int, payload: Dict) -> bool:
+        """
+        Update Device group
+        :param org_id: Organization ID
+        :param group_id: Group ID
+        :param payload: Dict of parameters to update on group
+        :return: Boolean of outcome
+        """
+        return self._call_api("PUT", f"{self.endpoint}/servergroups/{group_id}",
+                              params=self._org_param(org_id), json_data=payload)
+
+    def delete_group(self, org_id: int, group_id: int):
+        """
+        Delete Device group
+        :param org_id: Organization ID
+        :param group_id: Group ID
+        :return: Boolean of outcome
+        """
+        return self._call_api("DELETE", f"{self.endpoint}/servergroups/{group_id}", params=self._org_param(org_id))
+
+    # Vulnerability Sync
+    def upload_vulnerability_sync_file(self, org_id: int, file_content, filename):
+        with io.BytesIO(file_content) as file:
+            files = [
+                ('file', (filename, file, 'text/csv'))
+            ]
+
+            headers = {
+                'Authorization': f"Bearer {self.api_key}"
+            }
+
+            try:
+                response = requests.post(f"{self.endpoint}/orgs/{org_id}/tasks/patch/batches/upload",
+                                         files=files, headers=headers)
+
+                if response.status_code == 200:
+                    return response.json().get("id")
+                else:
+                    raise PluginException(cause="Failed to upload file to Vulnerability Sync",
+                                          assistance=f"Response code: {response.status_code}, Content: {response.content}")
+            except Exception as e:
+                raise PluginException(cause="Failed to upload file to Vulnerability Sync",
+                                      assistance=f"Review encoded CSV file and try again: {e}")
+
+    def get_vulnerability_sync_batches(self, org_id: int):
+        return self._page_results_data(f"{self.endpoint}/orgs/{org_id}/tasks/batches")
+
+    def get_vulnerability_sync_batch(self, org_id: int, batch_id: int):
+        return self._call_api("GET", f"{self.endpoint}/orgs/{org_id}/tasks/batches/{batch_id}")
+
+    def update_vulnerability_sync_batch(self, org_id: int, batch_id: int, action: str):
+        return self._call_api("POST", f"{self.endpoint}/orgs/{org_id}/tasks/batches/{batch_id}/{action}")
+
+    def get_vulnerability_sync_tasks(self, org_id: int, params: Dict):
+        return self._page_results_data(f"{self.endpoint}/orgs/{org_id}/tasks", params)
+
+    def update_vulnerability_sync_task(self, org_id: int, task_id: int, action: str):
+        return self._call_api("PATCH", f"{self.endpoint}/orgs/{org_id}/tasks/{task_id}", params={"action": action})
```

Comment:
```suggestion
    def get_vulnerability_sync_batches(self, org_id: int) -> bool:
        return self._page_results_data(f"{self.endpoint}/orgs/{org_id}/tasks/batches")

    def get_vulnerability_sync_batch(self, org_id: int, batch_id: int) -> bool:
        return self._call_api("GET", f"{self.endpoint}/orgs/{org_id}/tasks/batches/{batch_id}")

    def update_vulnerability_sync_batch(self, org_id: int, batch_id: int, action: str) -> bool:
        return self._call_api("POST", f"{self.endpoint}/orgs/{org_id}/tasks/batches/{batch_id}/{action}")

    def get_vulnerability_sync_tasks(self, org_id: int, params: Dict) -> bool:
        return self._page_results_data(f"{self.endpoint}/orgs/{org_id}/tasks", params)

    def update_vulnerability_sync_task(self, org_id: int, task_id: int, action: str) -> bool:
        return self._call_api("PATCH", f"{self.endpoint}/orgs/{org_id}/tasks/{task_id}", params={"action": action})
```
Assuming this is bool?